Client/Frontend/Browser/Tab.swift

```diff
@@ -679,3 +682,66 @@ class TabWebViewMenuHelper: UIView {
         }
     }
 }
+
+// MARK: - Brave Search
+
+extension Tab {
+    /// Call the api on the Brave Search website and passes the fallback results to it.
+    /// Important: This method is also called when there is no fallback results
+    /// or when the fallback call should not happen at all.
+    /// The website expects the iOS device to always call this method(blocks on it).
+    func injectResults() {
+        let dispatchGroup = DispatchGroup()
```

Comment:
No need for `DispatchGroup`?
Because the group is only ever entered once and it only leaves once.
Group is only useful if there is simultaneous requests happening afaik.